src/lib.rs

```diff
@@ -95,3 +97,9 @@ pub trait GenFn4<T>: Fn(&Seed, &Point4<T>) -> T {}
 impl<T, F> GenFn2<T> for F where F: Fn(&Seed, &Point2<T>) -> T {}
 impl<T, F> GenFn3<T> for F where F: Fn(&Seed, &Point3<T>) -> T {}
 impl<T, F> GenFn4<T> for F where F: Fn(&Seed, &Point4<T>) -> T {}
+
+pub trait Module<T> {
```

Comment:
I don't know about changing the name of Module to Source/ValueSource. Not all of these will 'source' a value. Some will modify their passed values, others won't modify the value, but modify the point instead.

It'd be nice to sort them as SourceModule, CombinerModule, ModifierModule, etc., but that's probably being too specific.